### PR TITLE
Add a simple script to deploy MIQ on minishift

### PR DESCRIPTION
**Pull request by agrare**
Adds a setup script to deploy ManageIQ on minishift using a single
command.

### CODE REVIEW

**Comment by agrare** [comment]
@miq-bot assign @carbonin

File: bin/setup [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+project=mynamespace
```
Comment (agrare) [inline bin/setup]: TODO - take this from an ENV var or command line then default to mynamespace

File: bin/setup [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+then
+	echo "You must start minishift"
+	exit 1
+fi
+
+# Make sure path to the oc binary is on our PATH
+eval $(minishift oc-env)
+
+# Login as admin
+res=$(oc login -u system:admin)
```
Comment (agrare) [inline bin/setup]: For some reason bash is trying to execute the stdout of every command unless I set to a variable...not sure if this is just my bash env

Comment (carbonin) [reply bin/setup]: That's strange ... but I also think `system:admin` was broken in a recent version of minishift. I thought we had to use the admin user addon now. Was that fixed?

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+res=$(oc describe scc anyuid | grep Users | awk '{print $2}' | grep miq-anyuid)
```
Comment (carbonin) [inline bin/setup]: This will return the result for all namespaces. I think this should be `grep "$project:miq-anyuid"`

Comment (carbonin) [reply bin/setup]: Also is it better to use `grep -q` here?

File: bin/setup [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+if [ -z "$res" ];
+then
+	echo "Creating SCC miq-anyuid..."
```
Comment (carbonin) [inline bin/setup]: Technically this is assigning the `anyuid` SCC to the service account called `miq-anyuid`.

File: bin/setup (hunk 1 above)
Comment (carbonin) [reply bin/setup]: I think an envvar is a good call for a first pass.

**Comment by carbonin** [comment]
Also I think the script name shouldn't be just `setup` as that is typically not an entrypoint, but rather pre-run tasks. (For example, `bin/setup` in manageiq doesn't start the server).

I also want to make to clear from the name of the script that this is only for minishift. So maybe `deploy_on_minishift` or `minishift_quickstart`?

File: bin/setup (hunk 3 above)
Comment (agrare) [reply bin/setup]: :+1: switching to that and checking the return code

File: bin/setup (hunk 4 above)
Comment (agrare) [reply bin/setup]: :+1: changed

File: bin/setup (hunk 1 above)
Comment (agrare) [reply bin/setup]: :+1: done
Get from env var first then default to myproject

File: bin/deploy_on_minishift [hunk 5]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+project=$PROJECT
+if [ -z "$project" ];
+then
+	project=myproject
+fi
```
Comment (carbonin) [inline bin/deploy_on_minishift]: Can we use something like this here?

```bash
 project="${PROJECT:-myproject}"
```

Comment (agrare) [reply bin/deploy_on_minishift]: I'm sure we can, bash isn't my first, second, or third language :)

**Comment by agrare** [comment]
@carbonin I assume this will require some tweaks to work on the hammer branch?

**Comment by carbonin** [comment]
I don't think so... All the sccs and service accounts are named the same. Is there anything jumping out at you that won't work in hammer?

**Comment by agrare** [comment]
Okay cool I assumed there was some difference on the setup.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 15db993f6b05f504d3e8b56818d0bb60be508de9
Author: Nick Carboni <ncarboni@redhat.com>
Date:   Fri Aug 31 14:14:13 2018 -0400

    Merge pull request #303 from agrare/add_simple_minishift_setup
    
    Add a simple script to deploy MIQ on minishift
    
    (cherry picked from commit d40b35489b7038e361fda855fb255e8ad834830e)
```

**Pull request merged**